pkg/types/conf.go

```diff
@@ -223,25 +223,24 @@ func CreateCNIRuntimeConf(args *skel.CmdArgs, k8sArgs *K8sArgs, ifName string, r
 	cniArgs := os.Getenv("CNI_ARGS")
 	if cniArgs != "" {
 		for _, arg := range strings.Split(cniArgs, ";") {
-			for _, keyval := range strings.Split(arg, "=") {
-				if len(keyval) != 2 {
-					logging.Errorf("CreateCNIRuntimeConf: CNI_ARGS %s %s %d is not recognized as CNI arg, skipped", arg, keyval, len(keyval))
-					continue
-				}
+			keyval := strings.Split(arg, "=")
+			if len(keyval) != 2 {
+				logging.Errorf("CreateCNIRuntimeConf: CNI_ARGS %s %s %d is not recognized as CNI arg, skipped", arg, keyval, len(keyval))
+				continue
+			}
 
-				envKey := string(keyval[0])
-				envVal := string(keyval[1])
-				isExists := false
-				for _, rtArg := range rt.Args {
-					if rtArg[0] == envKey {
-						isExists = true
-					}
-				}
-				if isExists != false {
-					logging.Debugf("CreateCNIRuntimeConf: add new val: %s", arg)
-					rt.Args = append(rt.Args, [2]string{envKey, envVal})
+			envKey := string(keyval[0])
+			envVal := string(keyval[1])
```

Comment:
This code could not parse `CNI_ARGS=ARGS1=ARG1VAL;ARG2=ARG2VAL1=ARG2VAL2` (well, current code does not support it, of course).

So how about that?

```
                        keyval := strings.Split(arg, "=")
                        if len(keyval) < 2 {
                                logging.Errorf("CreateCNIRuntimeConf: CNI_ARGS %s %s %d is not recognized as CNI arg, skipped", arg, keyval, len(keyval))
                                continue
                        }

                        envKey := string(keyval[0])
                        envVal := string(strings.Join(keyval[1:], "="))
```

Reply:
Fixed, PTAL. Used strings.SplitN(..., 2) so we'll end up with ["BLAH", "foo=bar"] for those cases.